### PR TITLE
WIP - 140574405 refactor objects sent to react

### PR DESCRIPTION
**Pull request by drod1000**

### CODE REVIEW

**Comment by drod1000** [comment]
@jdconrad89 @Laszlo-JFLMTCO @mollybrown 

I created a dashboard presenter and tried to eliminate some of the logic from the controller. Let me know what you all think

File: app/controllers/dashboard_controller.rb [hunk 1]
```diff
@@ -1,15 +1,7 @@
...
-      if(current_user.portfolio)
-        @user = current_user
-        @projects = @user.projects
-        @avatar = current_user.portfolio.avatar.url(:square)
-        @markdown_info = current_user.portfolio.markdown_info
-        @project_avatars = Project.avatar_urls(current_user)
-      else
-        @projects = []
-      end
+      @dashboard_presenter = DashboardPresenter.new(current_user)
```
Comment (Laszlo-JFLMTCO) [inline app/controllers/dashboard_controller.rb:4]: @drod1000 Nice!!! Welcome back our dear friend, PORO! :-)

**Comment by Laszlo-JFLMTCO** [comment]
@drod1000 I will double check locally, but the changes look good.

As a side note, I opened a new chore ticket: https://www.pivotaltracker.com/story/show/142684761 to stop tracking the avatar images. They seem to "spam" each of our PRs. Currently I am not aware of any reason to track them, especially as they are in the `public` folder, but let's confirm as a team and update `.gitignore` to keep PR change content focused to the actual work we are doing.

@jdconrad89 @mollybrown Have you guys ran into this issue as well? What do you think?

FYI: @neight-allen

**Comment by Laszlo-JFLMTCO** [comment]
@mollybrown @jdconrad89 @drod1000 What do you guys think of opening a ticket to better understand `Travis CI` integration here? Locally as @drod1000 pointed out there is one failing test for this PR, but here `Travis CI` is passing.

I attached the screenshot of my terminal running `rspec` locally:
image

File: app/views/dashboard/index.html.erb [hunk 2]
```diff
@@ -1,13 +1,8 @@
...
-  {prerender: true}) %>
-<% else %>
-  <%= react_component('Main', {user: current_user,
-    signed_in: 0},
-  {prerender: true}) %>
-<% end %>
+<%= react_component('Dashboard', {
+  user: current_user,
+  projects: @dashboard_presenter.projects,
+  portfolio: current_user.portfolio,
```
Comment (Laszlo-JFLMTCO) [inline app/views/dashboard/index.html.erb]: @drod1000 I think we could utilize here the PORO as well, you already wrote the method. So make this line: `portfolio: @dashboard_presenter.portfolio,`


FYI @mollybrown @jdconrad89

**Comment by jdconrad89** [comment]
@Laszlo-JFLMTCO @drod1000 I was wondering why those images kept showing up as well, good call on the chore adding it to .gitignore!

**Pull request closed without merging**